### ISSUE TITLE
Expand PGH003 / create another rule to only allow `# pyright: ignore[diagnosticRuleName]` comments

### ISSUE THREAD

**Issue by kkom** [astral-sh/ruff#13532]
[blanket-type-ignore](https://docs.astral.sh/ruff/rules/blanket-type-ignore/#blanket-type-ignore-pgh003) bans blanket `# type: ignore` comments, however that's not helpful for Pyright users.

The docs suggest to use `# type: ignore[diagnosticRuleName]` instead (which is allowed), which Pyright doesn't understand – it only accepts `# pyright: ignore[diagnosticRuleName]`

Would it be possible to expand this rule / create another one to only allow `# pyright: ignore[diagnosticRuleName]` comments?

### Pyright materials on `# type: ignore` vs `# pyright: ignore`

https://github.com/microsoft/pyright/issues/2877#issuecomment-1065299211

https://github.com/microsoft/pyright/blob/main/docs/comments.md#line-level-diagnostic-suppression

### Example

Incorrect diagnostic rule caught by Pyright:

<img width="1438" alt="Screenshot 2024-09-27 at 10 22 41" src="https://github.com/user-attachments/assets/1a794a98-bc99-47a0-bf98-8cef9c846569">

`# type: ignore[ruleThatDoesNotExist]` interpreted as `# type: ignore` by Pyright:

<img width="838" alt="Screenshot 2024-09-27 at 10 22 51" src="https://github.com/user-attachments/assets/2f005463-e77e-4925-a875-9b8c8397a2f9">

**Comment by kkom** [comment]
I guess we could have two rules:

One that bans blanket type ignore comments (regardless of whether it's the vanilla or pyright one) -- the one I referred to could be expanded like this

Another that bans the vanilla "type: ignore" comment, in favour of the pyright version.

That is, if you're happy to have some Pyright specific rules in Ruff!

**Comment by MichaReiser** [comment]
Extending `blanked-type-ignore` to also lint for `pyright: ignore` without a rule code seems reasonable. 

 > Another that bans the vanilla "type: ignore" comment, in favour of the pyright version -- that would need to be new I think

Isn't this covered by by using a combination of `blanked-type-ignore` (which flags all blank ignore codes) and running pyright (which flags a type error everywhere where someone uses `type: ignore[code]`)? Or does pyright ignore all codes when using `type: ignore[code]`?

**Comment by kkom** [comment]
Yeah, the problem is that Pyright treats both of these as a blanket ignore:

* type: ignore
* type: ignore[whateverTextHere]
